CDP4Dal/Session.cs

```diff
@@ -569,24 +603,39 @@ private async Task AfterReadOrWriteOrUpdate(IList<CDP4Common.DTO.Thing> things,
         /// <param name="operationContainer">
         /// The provided <see cref="OperationContainer"/> to write
         /// </param>
+        /// <param name="files">List of file paths for files to be send to the datastore</param>
         /// <returns>
         /// an await-able <see cref="Task"/>
         /// </returns>
-        public async Task Write(OperationContainer operationContainer)
+        public async Task Write(OperationContainer operationContainer, IEnumerable<string> files)

```

Comment:
should we not check here whether the filepaths exist, and if not throw a `filenotfoundexception` ?

Reply:
It will do so in the Dal.OperationContainerFileVerification when it tries to read the stream to compute the hash, but an extra check here is probably also a good idea.